account_bank_statement_clear_partner/models/account_bank_statement.py

```diff
@@ -0,0 +1,11 @@
+# Copyright <YEAR(S)> <AUTHOR(S)>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class AccountBankStatementClearPartner(models.Model):
+    _inherit = "account.bank.statement"
+
+    def clear_partners(self):
+        self.line_ids.filtered(lambda x: not x.journal_entry_ids and not x.account_id).write({"partner_id": False})
```

Comment:
You can't put such long lines in v12. Please reformat to PEP8 80 cols limit. On v13+, this is done by black, but it's not the case for this version.

Reply:
Also you may call the method programmatically from a multi-recordset, so you need to either:

1. Add `self.ensure_one()` before for getting a knowledgeable error.
2. Do it multi-record aware.

In this case, it's easier the second option:
```suggestion
        self.mapped("line_ids").filtered(lambda x: not x.journal_entry_ids and not x.account_id).write({"partner_id": False})
```

Reply:
Changed and tested